crates/pallets/starknet/src/lib.rs

```diff
@@ -207,6 +208,13 @@ pub mod pallet {
     pub(super) type Pending<T: Config> =
         StorageValue<_, BoundedVec<(Transaction, TransactionReceiptWrapper), MaxTransactions>, ValueQuery>;
 
+    /// The Starknet pallet storage items.
+    /// STORAGE
+    /// Current building block's transactions.
+    #[pallet::storage]
+    #[pallet::getter(fn state)]
+    pub(super) type State<T: Config> = StorageValue<_, StateCommitments, ValueQuery>;
```

Comment:
Not sure we should store anything in the storage indeed. 
It will be present in the block header digest. I see no reason to double store it.

If we compute it as we go along we will have to store it for sure, but we can kill the storage in on_finalize.

If we compute it in one go we can do everything in memory.